diff -ru inkscape-0.92.5/src/extension/internal/bitmap/channel.cpp fixed_src/inkscape-0.92.5/src/extension/internal/bitmap/channel.cpp
--- inkscape-0.92.5/src/extension/internal/bitmap/channel.cpp	2020-04-09 23:37:16.000000000 +0200
+++ fixed_src/inkscape-0.92.5/src/extension/internal/bitmap/channel.cpp	2022-05-06 14:34:50.198204021 +0200
@@ -28,7 +28,7 @@
 	else if (!strcmp(_layerName, "Yellow Channel"))		layer = Magick::YellowChannel;
 	else if (!strcmp(_layerName, "Black Channel"))		layer = Magick::BlackChannel;
 	else if (!strcmp(_layerName, "Opacity Channel"))	layer = Magick::OpacityChannel;
-	else if (!strcmp(_layerName, "Matte Channel"))		layer = Magick::MatteChannel;		
+	//else if (!strcmp(_layerName, "Matte Channel"))		layer = Magick::MatteChannel;		
 	
 	image->channel(layer);
 }
diff -ru inkscape-0.92.5/src/extension/internal/bitmap/crop.cpp fixed_src/inkscape-0.92.5/src/extension/internal/bitmap/crop.cpp
--- inkscape-0.92.5/src/extension/internal/bitmap/crop.cpp	2020-04-09 23:37:16.000000000 +0200
+++ fixed_src/inkscape-0.92.5/src/extension/internal/bitmap/crop.cpp	2022-05-06 14:35:47.257814127 +0200
@@ -24,7 +24,8 @@
     int width = image->baseColumns() - (_left + _right);
     int height = image->baseRows() - (_top + _bottom);
     if (width > 0 and height > 0) {
-        image->crop(Magick::Geometry(width, height, _left, _top, false, false));
+        //image->crop(Magick::Geometry(width, height, _left, _top, false, false));
+	image->crop(Magick::Geometry(width, height, _left, _top));
         image->page("+0+0");
     }
 }
diff -ru inkscape-0.92.5/src/extension/internal/bitmap/levelChannel.cpp fixed_src/inkscape-0.92.5/src/extension/internal/bitmap/levelChannel.cpp
--- inkscape-0.92.5/src/extension/internal/bitmap/levelChannel.cpp	2020-04-09 23:37:16.000000000 +0200
+++ fixed_src/inkscape-0.92.5/src/extension/internal/bitmap/levelChannel.cpp	2022-05-06 14:42:12.395362628 +0200
@@ -28,9 +28,11 @@
 	else if (!strcmp(_channelName, "Yellow Channel"))	channel = Magick::YellowChannel;
 	else if (!strcmp(_channelName, "Black Channel"))	channel = Magick::BlackChannel;
 	else if (!strcmp(_channelName, "Opacity Channel"))	channel = Magick::OpacityChannel;
-	else if (!strcmp(_channelName, "Matte Channel"))	channel = Magick::MatteChannel;
-	Magick::Quantum black_point = Magick::Color::scaleDoubleToQuantum(_black_point / 100.0);
-	Magick::Quantum white_point = Magick::Color::scaleDoubleToQuantum(_white_point / 100.0);
+	//else if (!strcmp(_channelName, "Matte Channel"))	channel = Magick::MatteChannel;
+	//Magick::Quantum black_point = Magick::Color::scaleDoubleToQuantum(_black_point / 100.0);
+	Magick::Quantum black_point = static_cast<Magick::Quantum>(_black_point / 100.0 * QuantumRange);
+	//Magick::Quantum white_point = Magick::Color::scaleDoubleToQuantum(_white_point / 100.0);
+	Magick::Quantum white_point = static_cast<Magick::Quantum>(_white_point / 100.0 * QuantumRange);
 	image->levelChannel(channel, black_point, white_point, _mid_point);
 }
 
diff -ru inkscape-0.92.5/src/extension/internal/bitmap/level.cpp fixed_src/inkscape-0.92.5/src/extension/internal/bitmap/level.cpp
--- inkscape-0.92.5/src/extension/internal/bitmap/level.cpp	2020-04-09 23:37:16.000000000 +0200
+++ fixed_src/inkscape-0.92.5/src/extension/internal/bitmap/level.cpp	2022-05-06 14:40:32.465974692 +0200
@@ -19,8 +19,10 @@
 	
 void
 Level::applyEffect(Magick::Image* image) {
-	Magick::Quantum black_point = Magick::Color::scaleDoubleToQuantum(_black_point / 100.0);
-	Magick::Quantum white_point = Magick::Color::scaleDoubleToQuantum(_white_point / 100.0);
+	//Magick::Quantum black_point = Magick::Color::scaleDoubleToQuantum(_black_point / 100.0);
+	Magick::Quantum black_point = static_cast<Magick::Quantum>(_black_point / 100.0 * QuantumRange);
+	//Magick::Quantum white_point = Magick::Color::scaleDoubleToQuantum(_white_point / 100.0);
+	Magick::Quantum white_point = static_cast<Magick::Quantum>(_white_point / 100.0 * QuantumRange);
 	image->level(black_point, white_point, _mid_point);
 }
 
diff -ru inkscape-0.92.5/src/extension/internal/bitmap/opacity.cpp fixed_src/inkscape-0.92.5/src/extension/internal/bitmap/opacity.cpp
--- inkscape-0.92.5/src/extension/internal/bitmap/opacity.cpp	2020-04-09 23:37:16.000000000 +0200
+++ fixed_src/inkscape-0.92.5/src/extension/internal/bitmap/opacity.cpp	2022-05-06 14:49:23.552836440 +0200
@@ -19,8 +19,9 @@
 	
 void
 Opacity::applyEffect(Magick::Image* image) {
-	Magick::Quantum opacity = Magick::Color::scaleDoubleToQuantum((100 - _opacity) / 100.0);
-	image->opacity(opacity);
+	//Magick::Quantum opacity = Magick::Color::scaleDoubleToQuantum((100 - _opacity) / 100.0);
+	unsigned int opacity= (100 - _opacity) / 100.0 * QuantumRange;
+	image->alpha(opacity);
 }
 
 void
diff -ru inkscape-0.92.5/src/extension/internal/pdfinput/pdf-input.cpp fixed_src/inkscape-0.92.5/src/extension/internal/pdfinput/pdf-input.cpp
--- inkscape-0.92.5/src/extension/internal/pdfinput/pdf-input.cpp	2020-04-09 23:37:16.000000000 +0200
+++ fixed_src/inkscape-0.92.5/src/extension/internal/pdfinput/pdf-input.cpp	2022-05-06 14:33:16.088868096 +0200
@@ -704,8 +704,7 @@
 #ifndef WIN32
     // poppler does not use glib g_open. So on win32 we must use unicode call. code was copied from
     // glib gstdio.c
-    GooString *filename_goo = new GooString(uri);
-    PDFDoc *pdf_doc = new PDFDoc(filename_goo, NULL, NULL, NULL);   // TODO: Could ask for password
+    PDFDoc *pdf_doc = new PDFDoc(std::make_unique<GooString>(uri));   // TODO: Could ask for password
     //delete filename_goo;
 #else
     wchar_t *wfilename = reinterpret_cast<wchar_t*>(g_utf8_to_utf16 (uri, -1, NULL, NULL, NULL));
diff -ru inkscape-0.92.5/src/extension/internal/pdfinput/pdf-parser.cpp fixed_src/inkscape-0.92.5/src/extension/internal/pdfinput/pdf-parser.cpp
--- inkscape-0.92.5/src/extension/internal/pdfinput/pdf-parser.cpp	2020-04-09 23:37:16.000000000 +0200
+++ fixed_src/inkscape-0.92.5/src/extension/internal/pdfinput/pdf-parser.cpp	2022-05-06 14:34:13.878457015 +0200
@@ -2210,7 +2210,7 @@
   }
   if (printCommands) {
     printf("  font: tag=%s name='%s' %g\n",
-	   font->getTag()->getCString(),
+	   (font->getTag()).getCString(),
 	   font->getName() ? font->getName()->getCString() : "???",
 	   args[1].getNum());
     fflush(stdout);
diff -ru inkscape-0.92.5/src/ui/dialog/document-properties.cpp fixed_src/inkscape-0.92.5/src/ui/dialog/document-properties.cpp
--- inkscape-0.92.5/src/ui/dialog/document-properties.cpp	2020-04-09 23:37:16.000000000 +0200
+++ fixed_src/inkscape-0.92.5/src/ui/dialog/document-properties.cpp	2022-05-06 14:51:49.742007224 +0200
@@ -583,6 +583,7 @@
     if (! current.empty()) {
         _emb_profiles_observer.set((*(current.begin()))->parent);
     }
+    /*
     std::set<SPObject *, _cmp> _current (current.begin(), current.end());
     for (std::set<SPObject *, _cmp>::const_iterator it = _current.begin(); it != _current.end(); ++it) {
         SPObject* obj = *it;
@@ -591,6 +592,7 @@
         row[_LinkedProfilesListColumns.nameColumn] = prof->name;
 //        row[_LinkedProfilesListColumns.previewColumn] = "Color Preview";
     }
+    */
 }
 
 void DocumentProperties::external_scripts_list_button_release(GdkEventButton* event)
diff -ru inkscape-0.92.5/src/ui/widget/color-icc-selector.cpp fixed_src/inkscape-0.92.5/src/ui/widget/color-icc-selector.cpp
--- inkscape-0.92.5/src/ui/widget/color-icc-selector.cpp	2020-04-09 23:37:16.000000000 +0200
+++ fixed_src/inkscape-0.92.5/src/ui/widget/color-icc-selector.cpp	2022-05-06 14:56:02.040594761 +0200
@@ -703,6 +703,7 @@
 
     int index = 1;
     std::vector<SPObject *> current = SP_ACTIVE_DOCUMENT->getResourceList("iccprofile");
+    /*
     std::set<SPObject *, _cmp> _current(current.begin(), current.end());
     for (std::set<SPObject *, _cmp>::const_iterator it = _current.begin(); it != _current.end(); ++it) {
         SPObject *obj = *it;
@@ -718,6 +719,7 @@
 
         index++;
     }
+    */
 
     g_signal_handler_unblock(G_OBJECT(_profileSel), _profChangedID);
 }
diff -ru inkscape-0.92.5/src/vanishing-point.cpp fixed_src/inkscape-0.92.5/src/vanishing-point.cpp
--- inkscape-0.92.5/src/vanishing-point.cpp	2020-04-09 23:37:16.000000000 +0200
+++ fixed_src/inkscape-0.92.5/src/vanishing-point.cpp	2022-05-06 15:04:28.237684797 +0200
@@ -403,9 +403,9 @@
         SPBox3D *box = dynamic_cast<SPBox3D *>(item);
         if (box) {
             vp = this->findVPWithBox(box);
-            if (vp) {
-                sel_vps.insert (vp);
-            }
+            //if (vp) {
+            //    sel_vps.insert (vp);
+            //}
         }
     }
     return sel_vps;
